### ISSUE TITLE
Support custom database options in addition to --datadir

### ISSUE THREAD

**Issue by bbyars** [bbyars/mountebank#591]
The --datadir CLI flag introduced a file-based database, but it runs into file locking issues at load. Rather than coupling to a specific database technology, it would be better to add custom support (analagous to the --formatter flag). The impostersRepository has now defined a JavaScript interface (even if it's quite clunky), allowing module inclusion.

**Comment by gePower** [comment]
Hi, can you assign me to the issue?

**Comment by bbyars** [comment]
Before you spend time on it, let me check on a parallel initiative first. I'd talked to someone else about this using a plugin model.

**Comment by gePower** [comment]
Perfect !

**Comment by jbadeau** [comment]
what about something like https://knexjs.org/?

**Comment by TetsujinOni** [comment]
I'll note that in extreme cases, --datadir will exhaust the filesystem it's running on when request-recording is enabled and a high number of requests are made in load-testing scenarios. A less incompressible backend for imposters could be useful independently of a generative imposter.

**Comment by bbyars** [comment]
Yep. There's a mongo db implementation in the works that should be a much
more robust solution. The --datadir did at least force creating the right
interface (and a painful refactoring from sync operations to async), but
having a pluggable persistence mechanism is the next step.
-Brandon
